### DISCUSSION TITLE
https issue

### DISCUSSION THREAD

**Discussion by ghost** [postalserver/postal#1870] [Help with using Postal] [answered]
ok so ive got postal working on http basically I'm using nethserver we use apache proxypass to forward postal.domain.tld to docker ip:5000 now my issue is if i enable https on proxy pass i get the login for postal but on login i get the standard error message and in the logs i get ``HTTP Origin header (https://postal.domain.tld) didn’t match request.base_url (http://postal.domain.tld)`` could someone give me some pointers or suggestions. were to look for info to solve the issue

**Comment by moazam1** [comment]
Had similar issue when I was using Apache. Use below code it should solve the issue:

```
<Location />
	ProxyPass http://127.0.0.1:5000/ retry=0
	ProxyPassReverse http://127.0.0.1:5000/
	ProxyPreserveHost On
	ProxyErrorOverride Off

</Location>
```

For nginx use below code:

```
location / {
        proxy_set_header X-Real-IP  $remote_addr;
        proxy_set_header Host $host;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto https;
        proxy_pass http://127.0.0.1:5000/;
    }
```

**Comment by ghost** [comment]
Thanks for the info ill find out were to modify the relevant config files

On 16 Feb 2022, at 11:40 pm, Moazam ***@***.******@***.***>> wrote:


location / {
        proxy_set_header X-Real-IP  $remote_addr;
        proxy_set_header Host $host;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto https;
        proxy_pass http://127.0.0.1:5000/;
    }

**Comment by willpower232** [comment]
If that doesn't help, please send the relevant log entries here

**Comment by ghost** [comment]
Because of the way nethserver is setup using templates I needed to create a custom template ie

<VirtualHost *:443> ServerName domain.tld RewriteEngine On SSLEngine On ProxyPassMatch ^/.well-known/acme-challenge/ ! ProxyPass http://172.17.0.1:5000/ retry=0 ProxyPassReverse http://172.17.0.1:5000/ ProxyPreserveHost On ProxyErrorOverride Off <Location "/"> <RequireAll> Require all granted SSLRequireSSL </RequireAll> </Location> </VirtualHost>

Now here is the funny part i tried adding the postal.conf and got an error with httpd not loading logs state

Feb 16 23:26:07 neth httpd[7668]: (98)Address already in use: AH00072: make_sock: could not bind to address 0.0.0.0:80

and

netstat -ltnp | grep -w ':80’

outputs

tcp6 0 0 :::80 :::* LISTEN 21923/caddy



so out of curiosity i tried going to https://postal.neth.ksatdesign.com.au and login worked so the issue is caddy (or rather apache) so i think a simple solution would be to have caddy use port 81 and not have proxypass interacting with it

**Comment by ghost** [comment, marked as answer]
ok weve found a way to get it working so we dont use caddy and create a postal.conf with

<VirtualHost *:443>
    ServerName postal.domain.tld
    RewriteEngine On
    SSLEngine On
    ProxyPassMatch ^/.well-known/acme-challenge/ !
    RequestHeader set X-Forwarded-Proto "https"
    ProxyPass / http://172.17.0.1:5000/
    ProxyPassReverse / http://172.17.0.1:5000/
    ProxyPreserveHost On
    ProxyErrorOverride Off
</VirtualHost>

<VirtualHost *:80>
    ServerName postal.domain.tld
    RewriteEngine On
    ProxyPassMatch ^/.well-known/acme-challenge/ !
    RewriteCond %{HTTPS} !=on
    RewriteRule (.*) https://%{SERVER_NAME}$1 [R,L]
</VirtualHost>
